### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by tvdijen**
This PR bumps the saml2 library to the latest version and bumps several dev dependencies
It also bumps the minimum required PHP-version to 7.3, which is in line with PR #115

### CODE REVIEW

File: src/SAML2/AuthnRequest.php
Comment (tvdijen) [reply src/SAML2/AuthnRequest.php]: Good catch!

**Review by thijskh — APPROVED** [review]
Looks good to me. Needs rebasing/squashing before merging. I'll leave it up to @MKodde and @Shaky212 to see at what time merging this fits in their SF5 upgrade work

**Comment by tvdijen** [comment]
Note that the upgrade of saml2 has changed the way Issuer and NameID work.
I can already tell that this will require some minor changes to the gateway-bundle as well

File: composer.json
Comment (MKodde) [inline composer.json:18]: Very cool to add `psalm`, but shouldn't this be a dev dependency?

**Review by MKodde — COMMENTED** [review]
Thanks for the support on this @tvdijen ! In our current effort to add SF5 & PHP8 support to the bundle. This was one of the tasks on our agenda. This comes  at a very convenient time. 

Your changes make a lot of sense. I must admit I have not yet test-driven them in a real life scenario but we will do so quite soon. If any functional issues come to the surface we will address them then and there. 

For now see my one psalm related comment. You added psalm, but I do not see an integration in the QA tooling. Maybe I'm looking at a dev-leftover?

File: composer.json
Comment (tvdijen) [reply composer.json:18]: That should have never ended up in this PR.. I used it to make sure I didn't introduce any bugs. Will remove it!

**Comment by tvdijen** [comment]
> I must admit I have not yet test-driven them in a real life scenario but we will do so quite soon. If any functional issues come to the surface we will address them then and there.

I have, and Gateway breaks due to the old use of string Issuer/NameID's. They are objects now and therefore any occurence of `$nameId` should be replaced with `$nameId->getValue()`...

See: https://github.com/simplesamlphp/saml2/blob/release-4.x/UPGRADING.md

**Pull request merged**
